### ISSUE TITLE
Related type in cs-field-group renders stringified representation

### ISSUE THREAD

**Issue by balinterdi** [cardstack/cardstack#484]
The example screenshot is from the tools dummy app (note how `time-unit` and `karma-type` objects are rendered):

![screenshot 2018-11-28 at 21 18 27](https://user-images.githubusercontent.com/5022/49179806-63fa8e00-f353-11e8-8f72-64ce4d4940ef.png)

**Comment by balinterdi** [comment]
@bagby You mentioned seeing this a while back – had you created an issue for it (if you had, feel free to close this)?

**Comment by balinterdi** [comment]
@ef4 @bagby I expect to pick this up next, any objections?

**Comment by balinterdi** [comment]
After some digging into code and thinking, I'm not sure if we want to build this into the framework (or, maybe a better question is what exactly we want to build into the framework).

Take the example of post.belongsTo('reading-time-units') which is rendered as `{{cs-field post "readingTimeUnits"}}` (at least that's what it compiles down to). The `cs-field` template checks if there's a field renderer for the type derived from the time unit object, which in that case returns `undefined`.

If the field renderer is not found, the value (`{{post.readingTimeUnits}}`) is rendered and that's why we see the stringified time unit object.

If we received the correct type information back for the time unit object, we'd need to provide a "time-unit" field renderer for a readable value to be rendered.

I can't think of a good way to introduce a convention so that the app developer doesn't have to provide a field renderer for related types. In one case, it could be the `title` property of the related record that we want to show. Take the "karma type" type, however, which is the other stringified value on screen. In that case, title alone doesn't suffice because it would just display "5 bad" (karma value + karma type) where we'd probably need "5 bad karma" so the name of the type also has to be included in the rendered representation.

We could also create an explicit meta property for this feature (work for this started in https://github.com/cardstack/cardstack/pull/456) but then the app developer would have to do some work just the same. If that's the case, providing field renderers for non-primitive type fields is a better (more intuitive) way to go.

What do you guys think?

**Comment by balinterdi** [comment]
My previous comment still stands but I've found that the app developer even has more flexibility when it comes to choosing a renderer for the field (within a `cs-field-group` or otherwise).

The way I've now fixed the rendering of time units (and comment karma) is as follows:

```handlebars
 {{#cs-field-group
    content
    name="reading-time"
    caption="Underestimated Read Time"
    fields="readingTimeValue,readingTimeUnit"
  as |group|}}
  <div data-test="reading-time">
    {{!--
      Only {{content.prop}} gets expanded to {{cs-field content "prop"}}
      So if a field is to be edited, it needs to use the `cs-field` helper otherwise
    --}}
    {{cs-field group "readingTimeValue"}}
    {{cs-field group.readingTimeUnit "title"}}
  </div>
{{/cs-field-group}}
```

With `{{cs-field group.readingTimeUnit "title"}}` the renderer used is the string-renderer (as the title field of time-units is a simple string so no editor needs to be provided.

One catch I came across is that writing `{{group.readingTimeUnit}}` will not render the corresponding "time-unit" editor (even if provided) as that expression is not compiled to `{{cs-field group "readingTimeUnit"`. This AST transform only works for `{{content.prop}}` (see comment in the hbs)
